StudentList/Fragments/FilterStudentsFragment.cs

```diff
@@ -53,7 +57,7 @@ public override void OnViewCreated(View view, Bundle savedInstanceState)
         {
             base.OnViewCreated(view, savedInstanceState);
 
-            if (this.studentFilter != default(StudentFilter))
+            if (this.studentFilter != StudentFilter.Default)
```

Comment:
You don't need a check here. If filter is a default filter, then you will simply have nulls in all fields, so nothing goes wrong if you set these nulls to textboxes.